[MA-0] Sprint1 developments.

cankurtur opened the pull request:
Fixed #8 #10 #13 #14 #15 #16 #17 #18 #19

Commit:
[MA-0] Sprint1 developments.

gidercagri commented on movieApp/movieApp/Scenes/Main/MainViewController.swift:
```diff
@@ -8,46 +8,75 @@
+
+    func scrollViewDidScroll(_ scrollView: UIScrollView) {
+        let scrollViewStartPoint = scrollView.contentOffset.y
+        let scrollViewYConstant = 143 - (scrollViewStartPoint + 143)
```
Constants

cankurtur replied:
fixed

gidercagri commented on movieApp/movieApp/Scenes/CardViewWithImageAndDetails/CardViewWithImageAndDetails.swift:
```diff
@@ -40,8 +41,13 @@ class CardViewWithImageAndDetails: UIView {
+        titleLabel.textColor = UIColor.almostBlack
+        detailsLabel.font = UIFont.body1
+        detailsLabel.alpha = Constants.detailsLabelOpacity
+        detailsLabel.textColor = UIColor.almostBlack
```
! Aşağıdaki configure metoduyla ilgili !
Tek bir if içinde kontrol edersek, herhangi bir değer eksik olursa tüm View çizilemez. Örnek olarak resim gelmedi backend tarafından, full beyaz boş bir View olur. İsteğimiz bu şekilde mi?

cankurtur replied:
Haklısın bu şekilde bir isteğimiz yok. If let kullanımını kaldırdım.

gidercagri commented on movieApp/movieApp/Scenes/CardViewWithImageAndDetails/CardViewWithImageAndDetails.swift:
```diff
@@ -40,8 +41,13 @@ class CardViewWithImageAndDetails: UIView {
 
     private func setupUI() {
-        self.backgroundColor = UIColor.white
         self.layer.cornerRadius = Constants.cornerRadius
```
no need to use self. in this case

cankurtur replied:
fixed

gidercagri commented on movieApp/movieApp/Scenes/CardViewWithImageAndDetails/TableViewCell/CustomTableViewCell.swift:
```diff
@@ -8,21 +8,50 @@
+        self.cardViewWithImageAndDetails.layer.masksToBounds = true
+
+        let bezierPath = UIBezierPath.init(roundedRect: self.viewContainer.bounds, cornerRadius: Constants.cornerRadius)
+        self.viewContainer.layer.shadowPath = bezierPath.cgPath
```
redundant self usage

cankurtur replied:
fixed

gidercagri commented on movieApp/movieApp/Scenes/Main/MainViewController.swift:
```diff
@@ -8,46 +8,75 @@
+    // MARK: - Business Logic
+    func setupUI() {
+        view.backgroundColor = UIColor.whiteTwo
+        headlineLabel.font = UIFont.headline2
```
No need to adjust font, text color in code. You can adjust on storyboard

cankurtur replied:
fixed

gidercagri commented on movieApp/movieApp/Scenes/Main/MainViewController.swift:
```diff
@@ -8,46 +8,75 @@
-        filmTableView.delegate = self
-        filmTableView.dataSource = self
+        setupUI()
+        setupUITableView()
```
setupUITableView and setupUIScrollView methods are not belong to extensions. They should be with class functions

cankurtur replied:
fixed

gidercagri reviewed (commented):
I have some comments

Commit:
[MA-0] Unnecessary Self usage removed.
[MA-0] Setup functions of TableView and ScrollView fixed.
[MA-0] Text properties given from Storyboard instead of swift file.

Commit:
[MA-0] If let usage deleted.

batuhanabay commented:
- AppIcon missing
- Parameter names should be camelCase
- We should not access any icon by using UIImage(named:"iconName"). There should be an extension for UIImage. Additionally, we shouldn't use String parameters for images like we did on ImageWithLabelUIModel.
- Theme related files should be gathered under a theme folder
- Redundant self usage should be checked and cleared
- Redundant type identifier usage should be checked and cleared **(Ex: backgroundColor = UIColor.white -> backgroundColor = .white)**
- All UI components should be tested with long/short text/images etc. to check constraint or component configuration like label's number of lines value etc.
- Generic components should be located under Components folder. (CardViewWithImageAndDetails component located under scenes in current situation)

batuhanabay commented on movieApp/movieApp/Scenes/Main/MainViewController.swift:
```diff
@@ -8,46 +8,66 @@
+// MARK: - UIScrollViewDelegate
+extension MainViewController: UIScrollViewDelegate {
+    func scrollViewDidScroll(_ scrollView: UIScrollView) {
+        let scrollViewStartPoint = scrollView.contentOffset.y
+        let scrollViewYConstant = Constants.topViewHeightToSafeArea - (scrollViewStartPoint + Constants.topViewHeightToSafeArea)
+
+        if scrollViewYConstant < Constants.limitForBarShowing {
+            title = Constants.titleBarText
+        } else {
+            title = Constants.largeTitleBarText
+        }
```
I could not understand the operation that you made. Could you tell it on our review meeting?

batuhanabay commented on movieApp/movieApp/Utility/Extensions/UIFontExtension.swift:
```diff
@@ -0,0 +1,26 @@
+    static let body2 = getSFText(.regular, size: 12)
+    static let point = getSFText(.black, size: 10)
+
+    public enum OpenSFProType: String {
```
I think we should not access these enum or method under UIFont type so we should separate these function to their related types.

batuhanabay reviewed (changes requested):
- AppIcon missing
- Parameter names should be camelCase
- We should not access any icon by using UIImage(named:"iconName"). There should be an extension for UIImage. Additionally, we shouldn't use String parameters for images like we did on ImageWithLabelUIModel.
- Theme related files should be gathered under a theme folder
- Redundant self usage should be checked and cleared
- Redundant type identifier usage should be checked and cleared (Ex: backgroundColor = UIColor.white -> backgroundColor = .white)
- All UI components should be tested with long/short text/images etc. to check constraint or component configuration like label's number of lines value etc.
- Generic components should be located under Components folder. (CardViewWithImageAndDetails component located under scenes in current situation)

Commit:
[MA-0] App icon added.

Commit:
[MA-0] Corner radius value of shadow layer added.

Commit:
[MA-0] Redundant type identifier deleted.

Commit:
[MA-0] UIImage, UIFont and UIColor extensions created and edited.
[MA-0] The files related to the theme are gathered under the theme folder.
[MA-o] Old UI Models changed according to newly created extensions.

Commit:
[MA-0] Localizable file created.

Commit:
[MA-0] Localizable file of Main added.
[MA-0] Redundant self usage deleted.
[MA-0] More understandable names given to variables.

Commit:
[MA-0] Unnecessary localizable files deleted.

Commit:
[MA-0] Status bar fixed.

Commit:
[MA-0] Parameter names fixed.

Commit:
[MA-0] Set function for UIColorExtension and UIImageExtension moved to related classes.
[MA-0] The files and folder related to the theme placed under the folder and organized.

Commit:
[MA-0] "private" keyword added to struct.

Commit:
[MA-0] CardViewWithImageAndDetails component moved to Components folder.
[MA-0] CustomTableViewCell.xib created and the customTableCell in MainViewController moved inside of the CustomTableViewCell that is newly created.

Commit:
[MA-0] Unnecessary math operations are deleted.
[MA-0] Top View fixed.

Commit:
[MA-0] Localizable feature added to UILabel, UIButton and String.
[MA-0] Localizable keys added for different language.

Commit:
[MA-0] Number of lines and minimum font size features adjusted.

batuhanabay reviewed (approved):
- Localizable stringler içerisinde aynı key iki kez kullanılmış ve TR dosyasında açıklamalar Ingilizce girilmiş, fişlenmeli.
- Colors.swift dosyasında default olarak .white dönüyor olmak eğer assetler içerisinde white özelinde bir değişiklik veya hata yapılırsa recursive çağrı ile sonsuz döngüye sebep olur gibi duruyor, kontrol edilmeli.
- ViewController üzerindeki kodlar kontrol edilip toparlanmalı bu konu üzerinde 5 dk konuşuruz yarınki dailyde.
- CardViewWithImageAndDetails.swift dosyasında configure methodu içerisinde coverImageView'a image setlerken ilgili image a text ile erişiyoruz, bunu güncelleyip Icon temasından çekmeliyiz.
